oeps/oep-0051-bp-conventional-commits.rst

```diff
@@ -0,0 +1,167 @@
+
+Commit message subjects should be short.  Put more information in the body of the commit message to fully explain your change.
+
+Body
+----
+
+The subject of the commit is rarely enough information to fully understand the commit.  The body can contain as much information as you like.  Be generous.  Take a moment to think about what you would want to know if someone else had authored this commit.
+
+The more information you can put in the body of the commit message, the better. It's not unreasonable to have two paragraphs of explanation in the body.
+
+Breaking changes especially should have detailed information about the implications and alternatives.
+
+
+Footer
+------
+
+We are not requiring any formalized footers in the commit message, though the use of  a ``BREAKING CHANGE:`` footer is encouraged as a way of providing more details about breaking changes.
+
+
+Discussion
+==========
+
+Conventional Commits asks us to categorize changes into a small number of categories.  There will naturally be cases where it isn't obvious which commit type to use.
+
+Choosing the commit type to use for a commit is important, but it's only the first step.  If you have doubts about which commit type to use, choose the highest-priority type that could apply.  Then write a detailed body explaining the full complexity of your change.
+
+**feat vs fix**: some user-visible changes to features could be classified as "feat" or "fix".  Choose "feat" if the change adds to the set of features.  Choose "fix" if the change affects how a feature behaves.  Yes, this is still subjective.  
+
+**breaking changes to features**: changing how a feature works is not a breaking change.  For example, users are sent to a new experience instead of the old experience. This is not a breaking change.  It should get a "feat" label, but not a "feat!" label.
+
+**DEPR**: removing deprecated code likely is a breaking change.
+
+**pinned dependencies**: updating the version of a pinned dependency seems like just a chore, but consider the repo consumer's perspective.  If an updated dependency adds a feature, then the one-line commit to update the pinned version should be marked "feat".
+
+If you are interested to see other discussion about these sorts of questions, the `Conventional Commits repo issues`__ have a number of threads.
+
+__ https://github.com/conventional-commits/conventionalcommits.org/issues
+
+
+Tooling
+=======
+
+One of the advantages of formalized commit messages is using them as input to tooling and conformance checkers.  We will investgate tooling in the future, and are making no recommendations now.
+
+
+Rationale
+=========
+
+The existing Conventional Commits standard is a familiar and widely adopted standard.  It's a good place to start our own guidelines.  It builds on our recently adopted pull request template to help focus developers on providing helpful information about their changes.
+
+We've adapted the guidelines to our own projects, and started small.
```

Comment:
Copying in or linking to an example of a real convential commit to an edX repo may be helpful to the reader.

Reply:
I've added one, let me know if you think it is a good one.

Reply:
Looks great!